File: leetcode/1.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
 var twoSum = function(nums, target) {
    let mp = {};
    for(let i=0;i<nums.length;i++) {
        let j = target - nums[i];
        if(typeof mp[j] !== "undefined" )return [mp[j],i];
        mp[nums[i]] = i;
    }
    return [];
};

File: go/L1/1.go
package L1

/*
	1.暴力 n^2
	2.哈希表 n
	97%
*/

func twoSum(nums []int, target int) []int {
	var mp = make(map[int]int)
	for i := 0; i < len(nums); i++ {
		if val, ok := mp[target-nums[i]]; ok {
			return []int{val, i}
		}
		mp[nums[i]] = i
	}
	return nil
}
